### PR TITLE
always use titandb statistics

### PR DESCRIPTION
**Pull request by tabokie**
- update for https://github.com/tikv/titan/pull/250
- initialize titandb statistics differently
- watch C bindings in build script

Signed-off-by: tabokie <xy.tao@outlook.com>

### CODE REVIEW

File: src/rocksdb_options.rs [hunk 1]
```diff
@@ -718,7 +718,12 @@ impl DBOptions {
 
     pub fn set_titandb_options(&mut self, opts: &TitanDBOptions) {
         unsafe {
+            let is_null_before = self.titan_inner.is_null();
             self.titan_inner = crocksdb_ffi::ctitandb_options_copy(opts.inner);
+            if is_null_before {
+                // Titan uses a different statistics.
+                self.enable_statistics(true);
```
[the remark this answers is not in the document]
Comment (tabokie) [reply src/rocksdb_options.rs:725]: The statistics in titan options will be overwrite in `ctitandb_options_set_rocksdb_options`.

[the remark this answers is not in the document]
Comment (BusyJay) [reply src/rocksdb_options.rs:725]: This is too subtle. Why not always use titan's version?

[the remark this answers is not in the document]
Comment (tabokie) [reply src/rocksdb_options.rs:725]: Even though `TitanDBOptions` physically contains RocksDB's options, it doesn't expose accessors for them.

[the remark this answers is not in the document]
Comment (BusyJay) [reply src/rocksdb_options.rs:725]: What I mean is always using `titandb::CreateDBStatistics()` in C no matter if it's Titan or not. So we don't need to worry about these corner cases.

[the remark this answers is not in the document]
Comment (tabokie) [reply src/rocksdb_options.rs:725]: Okay.

File: librocksdb_sys/crocksdb/c.cc [hunk 2]
```diff
@@ -2648,9 +2649,14 @@ void crocksdb_options_set_sst_partitioner_factory(
 }
 
 void crocksdb_options_enable_statistics(crocksdb_options_t* opt,
-                                        unsigned char v) {
+                                        unsigned char v,
+                                        unsigned char is_titandb) {
```
Comment (tabokie) [inline librocksdb_sys/crocksdb/c.cc:2653]:
```suggestion
                                        unsigned char v) {
```

File: librocksdb_sys/crocksdb/c.cc [hunk 3]
```diff
@@ -2648,9 +2649,14 @@ void crocksdb_options_set_sst_partitioner_factory(
...
   if (v) {
-    opt->rep.statistics = rocksdb::CreateDBStatistics();
+    if (is_titandb) {
+      opt->rep.statistics = rocksdb::titandb::CreateDBStatistics();
+    } else {
+      opt->rep.statistics = rocksdb::CreateDBStatistics();
+    }
```
Comment (tabokie) [inline librocksdb_sys/crocksdb/c.cc:2659]:
```suggestion
    opt->rep.statistics = rocksdb::titandb::CreateDBStatistics();
```

File: librocksdb_sys/crocksdb/crocksdb/c.h [hunk 4]
```diff
@@ -1078,7 +1078,7 @@ crocksdb_options_get_sst_partitioner_factory(crocksdb_options_t*);
 extern C_ROCKSDB_LIBRARY_API void crocksdb_options_set_sst_partitioner_factory(
     crocksdb_options_t*, crocksdb_sst_partitioner_factory_t*);
 extern C_ROCKSDB_LIBRARY_API void crocksdb_options_enable_statistics(
-    crocksdb_options_t*, unsigned char);
+    crocksdb_options_t*, unsigned char, unsigned char);
```
Comment (tabokie) [inline librocksdb_sys/crocksdb/crocksdb/c.h:1081]:
```suggestion
    crocksdb_options_t*, unsigned char);
```

File: librocksdb_sys/src/lib.rs [hunk 5]
```diff
@@ -783,7 +783,7 @@ extern "C" {
     pub fn crocksdb_options_set_wal_recovery_mode(options: *mut Options, mode: DBRecoveryMode);
     pub fn crocksdb_options_set_max_subcompactions(options: *mut Options, v: u32);
     pub fn crocksdb_options_set_wal_bytes_per_sync(options: *mut Options, v: u64);
-    pub fn crocksdb_options_enable_statistics(options: *mut Options, v: bool);
+    pub fn crocksdb_options_enable_statistics(options: *mut Options, v: bool, is_titandb: bool);
```
Comment (tabokie) [inline librocksdb_sys/src/lib.rs:786]:
```suggestion
    pub fn crocksdb_options_enable_statistics(options: *mut Options, v: bool);
```

File: src/rocksdb_options.rs [hunk 6]
```diff
@@ -718,7 +718,12 @@ impl DBOptions {
 
     pub fn set_titandb_options(&mut self, opts: &TitanDBOptions) {
         unsafe {
+            let is_null_before = self.titan_inner.is_null();
```
Comment (tabokie) [inline src/rocksdb_options.rs:721]:
```suggestion
```

File: src/rocksdb_options.rs [hunk 7]
```diff
@@ -718,7 +718,12 @@ impl DBOptions {
...
             self.titan_inner = crocksdb_ffi::ctitandb_options_copy(opts.inner);
+            if is_null_before && self.get_statistics().is_some() {
+                // Titan uses a different statistics.
+                self.enable_statistics(true);
+            }
```
Comment (tabokie) [inline src/rocksdb_options.rs:726]:
```suggestion
```

File: src/rocksdb_options.rs [hunk 8]
```diff
@@ -881,7 +886,11 @@ impl DBOptions {
 
     pub fn enable_statistics(&mut self, v: bool) {
         unsafe {
-            crocksdb_ffi::crocksdb_options_enable_statistics(self.inner, v);
+            crocksdb_ffi::crocksdb_options_enable_statistics(
+                self.inner,
+                v,
+                !self.titan_inner.is_null(),
+            );
```
Comment (tabokie) [inline src/rocksdb_options.rs:893]:
```suggestion
            crocksdb_ffi::crocksdb_options_enable_statistics(self.inner, v);
```

**Review by BusyJay — APPROVED** [review]
What I mean

**Pull request merged**